### ISSUE TITLE
not seeing duplicate runs

### ISSUE THREAD

**Issue by yonestar** [jmtyszka/bidskit#29]
Hi
Thanks for your recent updates!  There is a bug such that duplicate runs are not being caught.  For example:

Under `work/conversion/sub-M80395860/ses-2/`  I have the following files:

> m80395860--acute_pa_32ch_mb8_v01_r01--EP--23.json
m80395860--acute_pa_32ch_mb8_v01_r01--EP--23.nii.gz
m80395860--acute_pa_32ch_mb8_v01_r01--EP--25.json
m80395860--acute_pa_32ch_mb8_v01_r01--EP--25.nii.gz
m80395860--acute_pa_32ch_mb8_v01_r01_SBRef--EP--22.json
m80395860--acute_pa_32ch_mb8_v01_r01_SBRef--EP--22.nii.gz
m80395860--acute_pa_32ch_mb8_v01_r01_SBRef--EP--24.json
m80395860--acute_pa_32ch_mb8_v01_r01_SBRef--EP--24.nii.gz

Notice the duplicates.

In the output directory, under `bidskit_out/sub-M80395860/ses-2/func/`, I get only the following files:

> sub-M80395860_ses-2_task-acute_run-02_bold.json
sub-M80395860_ses-2_task-acute_run-02_bold.nii.gz
sub-M80395860_ses-2_task-acute_run-02_events.tsv
sub-M80395860_ses-2_task-acute_run-02_sbref.json
sub-M80395860_ses-2_task-acute_run-02_sbref.nii.gz

So it correctly sees that there are 2 runs, because it adds run-02.  But it does not copy over the run-01 files.

ALSO:  the run order is not preserved by bidskit.  I have Runs 23 and 25 in the coversion dir. Bidskit renames run 23 (not run 25) to become run-02.  It would greatly help if order was preserved (i.e., the earlier number in the sequence becomes run-01, the next number becomes run-02, etc.).

Thank you!!
Yoni (CU Boulder Wager lab)

**Comment by jmtyszka** [comment]
Thanks for spotting this Yoni! Really appreciate the level of detail on the issue. Looks like I regressed the run numbering issue with my new fix. I'll look at this today and will upload a fix asap.

**Comment by jmtyszka** [comment]
This should be fixed in the latest release (v1.1.1), but let me know if you continue to see any weird behavior in the presence of duplicate series descriptions. Here's the output from a conversion with a similar structure to the one above, showing the correct behavior (incrementing the inferred run number correctly for each duplicate).
```
------------------------------------------------------------
Pass 2 : Populating BIDS source directory
------------------------------------------------------------
    Creating new dataset_description.json

------------------------------------------------------------
Processing subject Sphere32
------------------------------------------------------------
  Processing session 1
  BIDS working subject directory : /Users/jmt/sandbox/bids/work/conversion/sub-Sphere32
  BIDS working session directory : /Users/jmt/sandbox/bids/work/conversion/sub-Sphere32/ses-1
  BIDS source subject directory  : /Users/jmt/sandbox/bids/source/sub-Sphere32
  BIDS source session directory  : /Users/jmt/sandbox/bids/source/sub-Sphere32/ses-1
  Organizing task-mb4_run-01_bold
    EPI detected
  Creating /Users/jmt/sandbox/bids/source/sub-Sphere32/ses-1/func/sub-Sphere32_ses-1_task-rest_run-01_events.tsv
  Populating BIDS source directory
    Copying FBIRN_SPHERE_32ch--task-mb4_run-01_bold--EP--5.nii.gz to sub-Sphere32_ses-1_task-rest_run-01_bold.nii.gz
    Creating new sub-Sphere32_ses-1_task-rest_run-01_bold.json
  Organizing task-mb4_run-01_bold
    EPI detected
  Creating /Users/jmt/sandbox/bids/source/sub-Sphere32/ses-1/func/sub-Sphere32_ses-1_task-rest_run-02_events.tsv
  Populating BIDS source directory
    Copying FBIRN_SPHERE_32ch--task-mb4_run-01_bold--EP--7.nii.gz to sub-Sphere32_ses-1_task-rest_run-02_bold.nii.gz
    Creating new sub-Sphere32_ses-1_task-rest_run-02_bold.json
  Organizing task-mb4_run-01_bold_SBRef
    EPI detected
  Creating /Users/jmt/sandbox/bids/source/sub-Sphere32/ses-1/func/sub-Sphere32_ses-1_task-rest_run-01_sbref.nii.gz
  Populating BIDS source directory
    Copying FBIRN_SPHERE_32ch--task-mb4_run-01_bold_SBRef--EP--4.nii.gz to sub-Sphere32_ses-1_task-rest_run-01_sbref.nii.gz (overwrite)
    Creating new sub-Sphere32_ses-1_task-rest_run-01_sbref.json
  Organizing task-mb4_run-01_bold_SBRef
    EPI detected
  Creating /Users/jmt/sandbox/bids/source/sub-Sphere32/ses-1/func/sub-Sphere32_ses-1_task-rest_run-02_sbref.nii.gz
  Populating BIDS source directory
    Copying FBIRN_SPHERE_32ch--task-mb4_run-01_bold_SBRef--EP--6.nii.gz to sub-Sphere32_ses-1_task-rest_run-02_sbref.nii.gz (overwrite)
    Creating new sub-Sphere32_ses-1_task-rest_run-02_sbref.json
  Preserving conversion directory
```
